common/lib/xmodule/xmodule/tests/helpers.py

```diff
@@ -34,18 +34,26 @@ class StubUserService(UserService):
     Stub UserService for testing the sequence module.
     """
 
-    def __init__(self, is_anonymous=False, **kwargs):
+    def __init__(self, is_anonymous=False, user=None, **kwargs):
         self.is_anonymous = is_anonymous
+        self.user = user
         super().__init__(**kwargs)
 
     def get_current_user(self):
         """
         Implements abstract method for getting the current user.
         """
         user = XBlockUser()
-        if self.is_anonymous:
+        if self.is_anonymous or (self.user and not self.user.is_authenticated):
```

Comment:
Since `is_anonymous` is only being passed in from one location we should update it to pass in a mock user instead of having two ways for specifying the user is anonymous.

Reply:
Ah yep, that makes sense. Before I saw this comment, I started to remove the `user` param to UserStubService in 124c2a0 , but prefer your suggestion, so will do that instead.